types/webpack-env/index.d.ts

```diff
@@ -103,21 +104,21 @@ declare namespace __WebpackModuleApi {
          * The data will be available at module.hot.data on the new module.
          * @param callback
          */
-        dispose<T>(callback: (data: T) => void): void;
+        dispose<T = any>(callback: (data: T) => void): void;
```

Comment:
Yeah, I just figured I shouldn't break the API for whoever is using the type arguments right now. But I guess I already spent that card to some degree with requiring TypeScript 2.3.